### DISCUSSION TITLE
Liferay: jQuery not found when upgrading from 7.0 to 13.0.1

### DISCUSSION THREAD

**Discussion by Sumair2000** [primefaces/community#2521] [PrimeFaces]
I have recently upgraded the PrimeFaces version from 7 to 13.0.1 and resolved the breaking changes. The portlets are working fine individually. However, when I place two PrimeFaces portlets on a single page. In each portlet, I have a javascript_include.xhtml file where I'm creating an alias for jQuery using noConflict, and this file is invoked in the header. Below is the code for each separate PrimeFaces portlet:

Including javascript_include.xhtml in the header:
```
<h:head>
    <ui:include src="include/javascript_include.xhtml" />
</h:head>
```

Portlet 1 javascript_include.xhtml:
```
<script type="text/javascript" src="/html/js/jquery/jquery.min.js?v=1.0"></script>
<script type="text/javascript" src="/html/js/jquery/jquery-ui.min.js?v=1.0"></script>

<script type="text/javascript">
    var jQuery_f_d = jQuery.noConflict();
    if (typeof jQuery_ns_connect == 'undefined') {
        jQuery_ns_connect = jQuery_f_d;
    }
</script>
```
Portlet 2 javascript_include.xhtml:
```
<script type="text/javascript" src="/html/js/jquery/jquery.min.js?v=1.0"></script>
<script type="text/javascript" src="/html/js/jquery/jquery-ui.min.js?v=1.0"></script>

<script type="text/javascript">
    var jQuery_announcement = jQuery.noConflict();
    var announcements_f = 'announcementForm';
    if (typeof jQuery_ns_connect == 'undefined') {
        jQuery_ns_connect = jQuery_announcement;
    }
</script>
```

When I place only one portlet on the page, it works fine, and there are no issues in the browser console. However, when I place another portlet on the same page, I encounter console errors like jQuery_f_d not found, announcements_f not found, and jQuery_ns_connect not found. Additionally, errors are appearing with PrimeFaces components, such as the date picker not functioning properly.

What could be the main issue, and what are your suggestions?

**Comment by melloware** [comment]
why are you including jquery on your page when PrimeFaces already includes it?  Sounds like you have a jQuery version conflict.

**Reply by Sumair2000** [reply]
I have already tried to removed that but the error remains same.

**Reply by melloware** [reply]
I don't use Portlets so I am not sure what your issue is other than it sounds like something is not configured correctly.  Maybe ask your Portlet provider?

**Reply by Sumair2000** [reply]
I am using Liferay 7.2 and have created PrimeFaces portlets. While there are no issues related to Liferay itself, I'm encountering errors when dropping two portlets on a single page. After thoroughly checking, I found that no other version of jQuery is being loaded, only 3.7.0 is being loaded.

**Reply by melloware** [reply]
Can you check with Liferay support or forums?   Most users here have probably not used LifeRay so your chances of getting an answer on Liferay forums is probably higher.

**Reply by Sumair2000** [reply]
I am already engaged with them, but there hasn't been a positive result yet. That's why I posted here, hoping someone can help out.

**Reply by melloware** [reply]
OK good. Well maybe someone here who also uses Liferay will chime in...

**Reply by Sumair2000** [reply]
@melloware Just to mention, when using PrimeFaces 7.0, it works perfectly fine. However, I started facing these issues after upgrading to 13.0.1. There might be some other breaking changes that are causing this.

**Reply by melloware** [reply]
@Sumair2000 did you ever get your LifeRay issue figured out?
